loadbalancer/algorithm.go

```diff
@@ -220,14 +221,56 @@ func hash(s string) uint32 {
 	return h.Sum32()
 }
 
-// Returns index of endpoint with closest hash to key's hash
-func (ch consistentHash) search(key string) int {
+// returns index in hash ring with the closest hash to key's hash
+func (ch consistentHash) searchRing(key string) int {
 	h := hash(key)
 	i := sort.Search(ch.Len(), func(i int) bool { return ch[i].hash >= h })
 	if i == ch.Len() { // rollover
 		i = 0
 	}
-	return ch[i].index
+
+	return i
+}
+
+// Returns index of endpoint with closest hash to key's hash
+func (ch consistentHash) search(key string) int {
+	ringIndex := ch.searchRing(key)
+	return ch[ringIndex].index
+}
+
+func computeLoadAverage(ctx *routing.LBContext) float64 {
+	sum := 1.0 // add 1 to include the request that just arrived
```

Comment:
Why do we need to start with 1?

Reply:
I.e. algo is looking for backend to forward this request why would it count it into the load?

Reply:
The paper said to include it, but the reason is that if you look at the load of the existing servers at any time it should be true that no server is overloaded by more than `(targetLoad*balanceFactor) +1` (+1 since we took the floor of targetLoad). If we don't include the incoming request, then that property may not hold after the incoming request is added to a backend server.